### PR TITLE
docs(AMQP): add API documentation

### LINKED ISSUE #14: Document connector AMQP API
As a platform developer, I want to know the events and data formats I need to send to connector via RabbitMQ when updating devices or publishing data. Besides, I also want to know which events are expected from connector.  

DoD:

- [x] Add `device.register`
- [x] Add `device.unregister`
- [x] Add `schema.update`
- [x] Add `data.publish`
- [x] Add `data.update`
- [x] Add `data.request`

### PR DESCRIPTION
**Pull request by netoax**
Changes:

- Add southbound messages (commands)
- Add northbound messages (control and measurements)

Closes #14

### CODE REVIEW

File: api/amqp.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    * `unit` **Number** sensor unit (V, A, W, W, etc)
+    * `typeId` **Number** data value type (boolean, integer, etc)
+    * `name` **String** sensor name
+
+#### Example
+
+```json
+{
+  "id": "fbe64efa6c7f717e",
+  "schema": {
```
Comment (netoax) [inline api/amqp.md]: (reminder-to-fix) it should be an array.

File: api/amqp.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
+# AMQP API
```
Comment (ramonhpr) [inline api/amqp.md]: About tree structure.
I'd prefer the folder name as `docs`, the name `api` can be referring to anything besides the specification of the API. What do you think?

File: api/amqp.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+## Message types (Southbound)
+
+### Update data
+
+Updates a thing's sensor value.
+
+#### Topic
+
+* `fog`
```
Comment (ramonhpr) [inline api/amqp.md]: i think that the topic is `fog-messages`

File: api/amqp.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+- Update thing's schema
+- Publish thing's data
+
+## Message types (Southbound)
+
+### Update data
+
+Updates a thing's sensor value.
+
+#### Topic
```
Comment (ramonhpr) [inline api/amqp.md]: As this is an amqp api document we should use the names defined by the protocol.
In other words:
topic -> exchange
key -> binding key

File: api/amqp.md [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+## Message types (Northbound)
+
+### Register thing
+
+Registers a thing on the cloud.
+
+#### Topic
+
+* `cloud`
```
Comment (ramonhpr) [inline api/amqp.md]: `cloud-messages`

**Review by ramonhpr — CHANGES_REQUESTED** [review]

File: api/amqp.md (hunk 2 above)
Comment (netoax) [reply api/amqp.md]: I agree with you.

File: api/amqp.md (hunk 3 above)
Comment (netoax) [reply api/amqp.md]: We assert the exchange only by using the `topic/exchange` name. The suffix `-messages` is only used when [asserting the message queue](https://github.com/CESARBR/knot-fog-connector/blob/master/src/network/AMQPConnection.js#L26) on message listening.

File: api/amqp.md (hunk 5 above)
Comment (netoax) [reply api/amqp.md]: Same comment above.

File: api/amqp.md (hunk 4 above)
Comment (netoax) [reply api/amqp.md]: Done.

File: api/amqp.md (hunk 2 above)
Comment (netoax) [reply api/amqp.md]: Done.

File: api/amqp.md (hunk 1 above)
Comment (netoax) [reply api/amqp.md]: Done.

File: api/amqp.md (hunk 3 above)
Comment (ramonhpr) [reply api/amqp.md]: Ack

File: api/amqp.md (hunk 5 above)
Comment (ramonhpr) [reply api/amqp.md]: Ack

**Review by ramonhpr — APPROVED** [review]

**Pull request merged**